File: GGAndroid_2.0/GGScanner/scanner-sample/src/main/java/io/scanner/sample/ScannerSampleActivity.java
package io.scanner.sample;

import android.content.Context;
import android.content.Intent;

import io.ganguo.library.bean.Constants;
import io.ganguo.scanner.bean.ScannerConfig;
import io.ganguo.vmodel.ViewModelActivity;
import io.scanner.sample.databinding.ActivityScannerSampleBinding;
import io.scanner.sample.viewmodel.ScannerSampleVModel;

/**
 * <p>
 * 扫描二维码/条形码 示例Demo
 * </p>
 * Created by leo on 2018/8/8.
 */
public class ScannerSampleActivity extends ViewModelActivity<ActivityScannerSampleBinding, ScannerSampleVModel> {
    /**
     * function：create Intent
     *
     * @param context
     * @param type
     * @return
     */
    public static Intent intentFor(Context context, @ScannerConfig.ScanType int type) {
        Intent intent = new Intent(context, ScannerSampleActivity.class);
        intent.putExtra(Constants.KEY_DATA, type);
        return intent;
    }

    @Override
    public ScannerSampleVModel createViewModel() {
        @ScannerConfig.ScanType
        int type = getIntent().getIntExtra(Constants.KEY_DATA, ScannerConfig.TYPE_ALL);
        return new ScannerSampleVModel(type);
    }

    @Override
    public void onViewAttached(ScannerSampleVModel viewModel) {

    }
}

File: GGAndroid_3.0/main/app-scanner-sample/src/main/java/io/scanner/sample/ScannerSampleActivity.kt
package io.scanner.sample

import android.content.Context
import android.content.Intent
import io.ganguo.scanner.bean.CodecType
import io.ganguo.viewmodel.core.view.ViewModelActivity
import io.scanner.sample.databinding.ActivityScannerBinding
import io.scanner.sample.viewmodel.ScannerSampleVModel

/**
 *
 *
 * 扫描二维码/条形码 示例Demo
 *
 * Created by leo on 2018/8/8.
 */
class ScannerSampleActivity : ViewModelActivity<ActivityScannerBinding, ScannerSampleVModel>() {

    override fun createViewModel(): ScannerSampleVModel {
        val type = intent.getIntExtra(KEY_DATA, CodecType.ALL.value)
        val supportScanningFrame = intent.getBooleanExtra(KEY_SUPPORT_FRAME, true)
        return ScannerSampleVModel(type, supportScanningFrame)
    }

    override fun onViewAttached(viewModel: ScannerSampleVModel?) {
    }

    companion object {
        private const val KEY_DATA = "data"
        private const val KEY_SUPPORT_FRAME = "mode"

        /**
         * function：create Intent
         *
         * @param context
         * @param type
         * @return
         */
        fun intentFor(context: Context, type: Int, supportScanningFrame: Boolean?): Intent {
            val intent = Intent(context, ScannerSampleActivity::class.java)
            intent.putExtra(KEY_DATA, type)
            intent.putExtra(KEY_SUPPORT_FRAME, supportScanningFrame)
            return intent
        }
    }

}
